### PR TITLE
Authentication and authorization

### PR DESCRIPTION
**Pull request by beval1**

### CODE REVIEW

File: monolith/src/main/java/com/society/server/dto/payload/SignupDTO.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.society.server.dto.payload;
+
+import com.society.server.utils.validators.EmailValidator;
+import jakarta.validation.constraints.NotBlank;
+import lombok.*;
```
Comment (petya0111) [inline monolith/src/main/java/com/society/server/dto/payload/SignupDTO.java:5]: Don't use lombok all and use specific imports.

**Review by petya0111 — COMMENTED** [review]
Fix lombok.* import

File: monolith/src/main/java/com/society/server/model/entity/BaseEntity.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@SuperBuilder
+//@Entity
+//@Inheri   tance(strategy=InheritanceType.TABLE_PER_CLASS)
+@MappedSuperclass
+public abstract class BaseEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
```
Comment (PeevG) [inline monolith/src/main/java/com/society/server/model/entity/BaseEntity.java:23]: We can use GenerationType.IDENTITY instead AUTO.
https://github.com/doctrine/dbal/issues/5614
"AUTO uses SEQUENCE by default on PostgreSQL. This seems sub-optimal as it forces INSERT queries to provide the id explicitly using nextval('sequence_name') which is not easy to do in ORM queries, and requires that you know the sequence name. On the other hand if it's set to IDENTITY it works better as it does it like MySQL's AUTO_INCREMENT, you can just omit the id field entirely."

**Review by PeevG — APPROVED** [review]

**Review by petya0111 — APPROVED** [review]
Great work with the fundamentals of the Backend 🙏💯✔️

File: monolith/src/main/java/com/society/server/model/entity/BaseEntity.java (hunk 2 above)
Comment (beval1) [reply monolith/src/main/java/com/society/server/model/entity/BaseEntity.java:23]: Good point

File: monolith/src/main/java/com/society/server/dto/payload/SignupDTO.java (hunk 1 above)
Comment (beval1) [reply monolith/src/main/java/com/society/server/dto/payload/SignupDTO.java:5]: I don't think that's important, intellij does it automatically when imports are >=5

**Pull request merged**